students/junyoung/westagram/users/views.py

```diff
@@ -0,0 +1,49 @@
+import json
+import re
+
+from django.views import View
+from django.http import JsonResponse
+
+from users.models import User
+
+class SignUpView(View):
+    def post(self,request):
+        try:
+            data                    = json.loads(request.body)
+            email                   = data['email']
+            password                = data['password']
+            phone_number            = data['phone_number']
+            email_validation        = r'\w+[@]+\w+[.]+\w+'
+            password_validation     = r'^(?=.*[a-zA-Z])(?=.*\d)(?=.*\W).{8,}'
+            phone_number_validation = r'^(010)\d{8}$'
+            
+            if not email or not password:
+                raise KeyError
+            
+            elif User.objects.filter(email = email).exists():
+                return JsonResponse({"message" : "DUPLICATED_EMAIL"}, status = 409)
+            
+            elif not re.match(email_validation, email):
+                return JsonResponse({"message" : "INVALID_EMAIL"}, status = 400)
+                    
+            elif not re.match(password_validation, password):
+                return JsonResponse({"message" : "INVALID_PASSWORD"}, status = 400)
+        
+            elif not re.match(phone_number_validation, phone_number):
+                return JsonResponse({"message" : "INVALID_PHONE_NUMBER"}, status = 400)
```

Comment:
검증 절차의 순서를 수정하는 것이 좋아 보입니다.

이유는 View에서 input에 대한 검증이 먼저되고 나서 사용자의 존재 여부(`User.objects.filter(email = email).exists()`)를 확인하셔야 Database에 불필요한 호출을 줄일 수 가 있겠습니다.

```suggestion
            elif not re.match(email_validation, email):
                return JsonResponse({"message" : "INVALID_EMAIL"}, status = 400)
                    
            elif not re.match(password_validation, password):
                return JsonResponse({"message" : "INVALID_PASSWORD"}, status = 400)
        
            elif not re.match(phone_number_validation, phone_number):
                return JsonResponse({"message" : "INVALID_PHONE_NUMBER"}, status = 400)
                
            elif User.objects.filter(email = email).exists():
                return JsonResponse({"message" : "DUPLICATED_EMAIL"}, status = 409)                
```